deploy/use-paymaster.ts

```diff
@@ -35,17 +35,39 @@ export default async function (hre: HardhatRuntimeEnvironment) {
 
   const erc20 = getToken(hre, emptyWallet);
 
+  const gasPrice = await provider.getGasPrice();
+
+  // Estimate gas fee for mint transaction
+  const gasLimit = await erc20.estimateGas.mint(emptyWallet.address, 100, {
+    customData: {
+      ergsPerPubdata: utils.DEFAULT_ERGS_PER_PUBDATA_LIMIT,
+      paymasterParams: {
+        paymaster: PAYMASTER_ADDRESS,
+        paymasterInput: '0x',
+      },
+    },
+  });
+
+  const fee = gasPrice.mul(gasLimit.toString());
+
   // Encoding the "ApprovalBased" paymaster flow's input
   const paymasterParams = utils.getPaymasterParams(PAYMASTER_ADDRESS, {
     type: 'ApprovalBased',
     token: TOKEN_ADDRESS,
-    minimalAllowance: ethers.BigNumber.from(1),
+    // set minimalAllowance as calculated fee
+    minimalAllowance: fee,
```

Comment:
`minimalAllowance` is the minimal allowance of the ERC20 token that the user should provide to the paymaster. For the testnet paymaster (the one that is provided by the Matter Labs on the testnet) indeed requires the 1-1 allowance between ETH and ERC20 tokens.

In this particular tutorial, however, the paymaster accepts just 1 unit of the token to pay for any fee, so `ethers.BigNumber.from(1)` should work here

Reply:
cool, thanks for the explanation. I'll include that as a comment.

Reply:
fixed in 5539d82b2ab0a43972745f063077a259610c9ef1